app/controllers/pets_controller.rb

```diff
@@ -19,12 +19,20 @@ def edit
 
   def update
     pet = Pet.find(params[:id])
-    pet.update(pet_params)
+    if !pet.update(pet_params)
+      flash[:notice] = "Failed to update pet: #{pet.errors.full_messages.to_sentence}"
```

Comment:
- `errors` returns the errors of an object that wasn't able to be added to the DB
- `full_messages` converts the errors into a readable array of errors
- `to_sentence` converts that array into one string of a sentence

Reply:
(rails magic)